### PR TITLE
pacman: fix upgrade: yes

### LINKED ISSUE #4274: `pacman --sync --sysupgrade` broken
### Summary

[This commit](https://github.com/ansible-collections/community.general/commit/1580f3c2b4876aeb06a0c785a14c5a4e057e843e#diff-ba1a07866081b3bf3a75e76b1c3154a4f9ca26e9a39915b8f491f0a5922f8f50R413) introduced a `-` where none should be. See [here](https://archlinux.org/pacman/pacman.8.html)

### Issue Type

Bug Report

### Component Name

pacman

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.12.2]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/marcus/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/marcus/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.2 (main, Jan 15 2022, 19:56:27) [GCC 11.1.0]
  jinja version = 3.0.3
  libyaml = True

```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
# /usr/lib/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 4.4.0

```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

archlinux

### Steps to Reproduce

```yaml (paste below)
How to reproduce:

- hosts: all
  tasks:
  - name: pacman upgrade
    pacman:
      upgrade: yes
```

### Expected Results

I expected my archlinux machines to be upgraded.

### Actual Results

```console (paste below)
fatal: [example.com]: FAILED! => {"attempts": 10, "changed": false, "msg": "Could not upgrade", "packages": ["python-pyparsing", "expat", "bind", "libxml2", "mariadb-libs", "tree"], "stderr": "/usr/bin/pacman: unrecognized option '--sys-upgrade'\n", "stderr_lines": ["/usr/bin/pacman: unrecognized option '--sys-upgrade'"]}

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by jraby**
Fixes #4274

Sorry about that, I tested the upgrade in check mode only apparently... (and even the test were wrong :facepalm: )

### CODE REVIEW

**Comment by felixfontein** [comment]
Thanks for the quick fix! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: changelogs/fragments/4275-pacman-sysupgrade.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - pacman - Fix ``upgrade: yes``.
+    (https://github.com/ansible-collections/community.general/pull/4275, https://github.com/ansible-collections/community.general/issues/4274)
```
[the remark this answers is not in the document]
Comment (jraby) [reply changelogs/fragments/4275-pacman-sysupgrade.yml]: i should really wait a few minutes before jumping in and trying to fix the test failures, you always provide good answers while I'm working on it and I see it after I push!

[the remark this answers is not in the document]
Comment (felixfontein) [reply changelogs/fragments/4275-pacman-sysupgrade.yml]: I'm not always around ;-) The remainder of my suggestion is still valid, the fragments should start lower-case after the dash, and end with a period.

[the remark this answers is not in the document]
Comment (jraby) [reply changelogs/fragments/4275-pacman-sysupgrade.yml]: should be good to go now :-)

**Review by felixfontein — APPROVED** [review]

**Pull request merged**